### PR TITLE
MINOR: improve StateStore JavaDocs

### PR DESCRIPTION
**Pull request by mjsax**
Clarify that state directory must use `storeName`

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by mjsax** [comment]
Call for review @guozhangwang @bbejeck @dguy 

This can go into `1.0.x` and `0.11.0.x`, too.

**Comment by asfgit** [comment]
FAILURE
 7973 tests run, 5 skipped, 1 failed.
--none--

**Comment by asfgit** [comment]
FAILURE
 7973 tests run, 5 skipped, 1 failed.
--none--

**Comment by asfgit** [comment]
FAILURE
 7973 tests run, 5 skipped, 1 failed.
--none--

File: streams/src/main/java/org/apache/kafka/streams/processor/StateStore.java [hunk 1]
```diff
@@ -20,13 +20,17 @@
 
 /**
  * A storage engine for managing state maintained by a stream processor.
- *
+ * <p>
+ * If the store is implemented as a persistent store, it <em>must</em> use the store name as directory to write to disk.
+ * Thus, the directory for the store is {@code <state.dir>/<application.id>/<taskId>/<storeName>}.
```
Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/processor/StateStore.java]: Do we want to expose the pattern of `<state.dir>/<application.id>/<taskId>`? Could we just say users must create the directory with the name equal to the state store name under the directory given by `ProcessorContext#stateDir()`?

Comment (bbejeck) [reply streams/src/main/java/org/apache/kafka/streams/processor/StateStore.java]: I agree with @guozhangwang on this, as if we change our approach we'd have to re-educate users on the naming convention.

**Comment by mjsax** [comment]
Updates this.

**Review by bbejeck — COMMENTED** [review]
Thanks @mjsax LGTM

**Comment by guozhangwang** [comment]
LGTM. Merged to trunk, 1.0, 0.11.0.

**Comment by asfgit** [comment]
SUCCESS 
 8081 tests run, 5 skipped, 0 failed.
--none--

**Comment by asfgit** [comment]
SUCCESS 
 8081 tests run, 5 skipped, 0 failed.
--none--

**Pull request closed without merging**